### PR TITLE
Reader: add foundations for Cold Start UI

### PR DESCRIPTION
**Pull request by bluefuton**
Cold Start will help new users of the Reader (and potentially existing ones with no subscriptions) find interesting things to subscribe to:

screen shot 2016-04-20 at 14 25 24

This PR adds routing and basic templates for v1, which will offer a choice of recommended sites and posts, but no search.

cc @jancavan

### CODE REVIEW

File: client/reader/start/index.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { connect } from 'react-redux';
+
+// Internal dependencies
+import Main from 'components/main';
+import Button from 'components/button';
+import Card from 'components/card';
+
+const debug = debugModule( 'calypso:reader:start' ); //eslint-disable-line no-unused-vars
+
+const ReaderStart = React.createClass( {
```
Comment (mtias) [inline client/reader/start/index.jsx]: Since you are just starting this, the folder should be named `reader-start` :)

Comment (bluefuton) [inline client/reader/start/index.jsx]: Fixed! Thanks @mtias :)

**Comment by bluefuton** [comment]
- [ ] Add Redux plumbing (no API endpoint yet)
- [ ] Create `SiteCard` and `PostCard` components and basic styling
- [ ] Add follow button to cards

File: client/state/reader/start/reducer.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * Tracks all known list objects, indexed by list ID.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = {}, action ) {
+	switch ( action.type ) {
+		case READER_START_RECOMMENDATIONS_RECEIVE:
+			return Object.assign( {}, state, keyBy( action.recommendations, 'site_ID' ) );
```
Comment (blowery) [inline client/state/reader/start/reducer.js]: probably want a new tree sections for reader sites and posts? do we start that move now?

Comment (bluefuton) [inline client/state/reader/start/reducer.js]: Yep, I thought the same. `/start` is pretty self-contained at the moment, so we shouldn't face any of the same issues when we had when moving the lists Flux stores across to Redux.

I'll create site and posts in the Redux tree, and have them consume the meta from `READER_START_RECOMMENDATIONS_RECEIVE`.

**Comment by bluefuton** [comment]
Going to split this one up and close it:

#5240 - Reader: add controller and route for /read/start
#5257 - Reader: add start recommendations to Redux tree

...and another PR to be added for the front-end components after #5240 ships.

**Pull request closed without merging**